src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -2225,6 +2226,11 @@ class run_impl
   {
     get_arg_setter()->set_arg_value(arg, bo);
 
+    auto bh = xrt_core::bo_int::get_buffer_handle(bo);
+    auto off = xrt_core::bo_int::get_offset(bo);
+    auto sz = bo.size();
+    cmd->get_exec_bo()->bind_at(arg.index(), bh, off, sz);
```

Comment:
Hmm, maybe this should be a kernel_command member function?   Keeps this function simpler and moves the responsibility to the object that owns the exec bo.
`cmd->bind_arg_at_index(arg.index(), bo);`

Reply:
Sounds good. I'll make the change.